dcpy/connectors/edm/recipes.py

```diff
@@ -140,9 +140,11 @@ def plan_recipe(recipe_path: Path) -> Recipe:
     # Determine the recipe version
     if recipe.version is None and recipe.version_strategy is not None:
         if recipe.version_strategy == VersionStrategy.bump_latest_release:
+            if recipe.version_type is None:
+                raise Exception("Recipe needs a 'version_type' to bump")
             prev_version = publishing.get_latest_version(recipe.product)
             recipe.version = versions.bump(
-                prev_version, bumped_part=recipe.version_type
+                prev_version, bumped_part=str(recipe.version_type)
```

Comment:
If you use an `else` after the exception, will mypy know that this is a `str` rather than `Optional[str]`? I assume that's what the casting is converting it from here. Would find that preferable personally, just to not have to cast

Reply:
@fvankrieken Well I'll be. It does. Good job `mypy`!